src/components/Auth/AuthForm.tsx

```diff
@@ -0,0 +1,201 @@
+import { upperFirst } from '@mantine/hooks';
+import { useForm } from '@mantine/form';
+import {
+  TextInput,
+  PasswordInput,
+  Text,
+  Paper,
+  Group,
+  PaperProps,
+  Button,
+  Divider,
+  Checkbox,
+  Anchor,
+  Stack,
+  Container,
+} from '@mantine/core';
+import { useTranslation } from 'react-i18next';
+import { GoogleButton } from 'components/shared/AuthButton/GoogleButton';
+import { GitHubButton } from 'components/shared/AuthButton/GitHubButton';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  createUserWithEmailAndPassword,
+  getAuth,
+  signInWithEmailAndPassword,
+} from '@firebase/auth';
+import { setRole, setUser } from 'redux/slice/userSlice';
+import { useDispatch } from 'react-redux';
+
+interface AuthenticationFormProps extends PaperProps {
+  type: 'login' | 'register';
+}
+export function AuthenticationForm({
+  type: formType,
+  ...props
+}: AuthenticationFormProps) {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const [type, setType] = useState(formType);
+
+  useEffect(() => {
+    setType(formType);
+  }, [formType]);
+
+  const form = useForm({
+    initialValues: {
+      email: '',
+      name: '',
+      password: '',
+      terms: true,
+      login: '',
+    },
+
+    validate: {
+      email: val => (/^\S+@\S+$/.test(val) ? null : 'Invalid email'),
+      password: val =>
+        val.length <= 6 ? 'Password should include at least 6 characters' : null,
+    },
+  });
+
+  const dispatch = useDispatch();
```

Comment:
use [useAppDispatch](https://redux-toolkit.js.org/usage/usage-with-typescript#getting-the-dispatch-type)